### PR TITLE
Implement 3D and transposed 3D convolutions.

### LINKED ISSUE #1278: Conv3d feature request
Hello! Are any of 3d cv layers (conv3d, maxpool3d, etc..) planned?

### PR DESCRIPTION
**Pull request by booti386**
Hi!
Please review this PR carefully, I started working both on neural networks and this crate only 4 days ago.
I hope there are no major mistakes.

### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Fixes #1278: Conv3d feature request

### Changes

Implement 3D and 3D transposed convolutions for all the supported backends.

### CODE REVIEW

**Comment by antimora** [comment]
Super cool!

Also I am glad you have added ONNX for Conv3d and implemented new ModuleDisplay!

I can review the PR but @nathanielsimard, @laggui and @louisfd are more qualified.

**Comment by nathanielsimard** [comment]
Thanks a lot @booti386 for the PR. @laggui can you check if the ONNX part is valid, and @louisfd can also validate kernel implementations.

**Review by louisfd — APPROVED** [review]
Looking good on my end!

**Review by antimora — APPROVED** [review]
I checked the import parts and they look good.

**Comment by antimora** [comment]
Can we add a couple check boxes under: Conv3d and ConvTranspose3d in https://github.com/tracel-ai/burn/blob/main/crates/burn-import/SUPPORTED-ONNX-OPS.md? Folks would be happy to see we now support Conv2d and ConvTranspose3d!

File: crates/burn-import/src/onnx/op_configuration.rs [hunk 1]
```diff
@@ -97,6 +99,63 @@ pub fn conv2d_config(curr: &Node) -> Conv2dConfig {
...
+
+    // the channels are inverted in the weight tensor
+    let shape = weight.shape.clone().unwrap();
+    let channels: [usize; 2] = [shape[1], shape[0]];
+
+    for (key, value) in curr.attrs.iter() {
+        match key.as_str() {
+            "kernel_shape" => kernel_shape = value.clone().into_i64s(),
+            "strides" => strides = value.clone().into_i64s(),
+            "pads" => pads = value.clone().into_i64s(),
```
Comment (laggui) [inline crates/burn-import/src/onnx/op_configuration.rs:127]: Just realized we don't parse the `auto_pad` attribute (mutually exclusive) but that applies to all conv so no worries we can keep it as the current limitation. Don't think it's used that much.

**Review by laggui — APPROVED** [review]
Looked over the ONNX import, everything looks good! 👍 

Thanks for adding conv3d 🙏

**Comment by laggui** [comment]
> Great! Thank you all for your very positive reviews!

Before we merge could you update the [SUPPORTED-ONNX-OPS.md](https://github.com/tracel-ai/burn/blob/main/crates/burn-import/SUPPORTED-ONNX-OPS.md) file to reflect the new conv3d (as pointed out by @antimora)?

Thank you!

**Review by antimora — APPROVED** [review]
LGTM

**Comment by antimora** [comment]
A test is failing after a recent Async merge:

image

Since this is not related to this PR changes, @nathanielsimard @louisfd @laggui , how should we go about this?

**Comment by laggui** [comment]
The issue has been fixed on main.

Pretty sure the tests were all passing before the rebase so we could probably just merge, but if not sure then the fix could be merged before running the tests.

**Pull request merged**